### PR TITLE
Frankenporter

### PR DESCRIPTION
**Pull request by jdent02**

### CODE REVIEW

File: ui/objects.py [hunk 1]
```diff
@@ -162,7 +162,7 @@ class ASOBJECT_PT_export_override(bpy.types.Panel):
     @classmethod
     def poll(cls, context):
         renderer = context.scene.render
-        return renderer.engine == 'APPLESEED_RENDER' and context.object is not None and context.object.type in {'MESH', 'CURVE', 'SURFACE', 'EMPTY'}
+        return renderer.engine == 'APPLESEED_RENDER' and context.object is not None and context.object.type == 'EMPTY'
```
Comment (dictoon) [inline ui/objects.py:165]: What's the purpose of this change?

File: translators/translator.py [hunk 2]
```diff
@@ -34,146 +34,30 @@
...
-        :return:
-        """
-
-        raise NotImplementedError()
+    def flush_entities(self, as_scene, as_main_assembly, as_project):
+        raise NotImplementedError
 
     @staticmethod
-    def _convert_matrix(m):
-        """
```
Comment (dictoon) [inline translators/translator.py:144]: Why did you remove all these comments?

File: render/__init__.py [hunk 3]
```diff
@@ -74,13 +74,10 @@ class RenderAppleseed(bpy.types.RenderEngine):
     bl_idname = 'APPLESEED_RENDER'
     bl_label = 'appleseed'
     bl_use_preview = True
-    bl_use_shading_nodes = True
+    # bl_use_shading_nodes = True
```
Comment (dictoon) [inline render/__init__.py:77]: Remove?

File: render/__init__.py [hunk 4]
```diff
@@ -96,27 +93,19 @@ def __init__(self):
...
     #
     # Destructor.
     #
 
     def __del__(self):
-        try:
-            self.__stop_rendering()
-        except:
-            pass
+        logger.debug("Deleting render engine")
```
Comment (dictoon) [inline render/__init__.py:102]: Elsewhere you seem to prefix messages with `appleseed: `, that sounds like a good idea, do it here too (and make sure it's done for all log messages)?

File: translators/cameras/final.py [hunk 5]
```diff
@@ -36,52 +37,63 @@
...
+        self.__as_cam_params = self.__get_cam_params(depsgraph.scene_eval, engine)
 
-        self.__as_camera.set_parameters(as_camera_params)
+        self.__as_camera.set_parameters(self.__as_cam_params)
 
-    def set_xform_step(self, time):
-        self.__xform_seq.set_transform(time,
-                                       self._convert_matrix(self.__engine.camera_model_matrix(self.bl_camera)))
+    def flush_entities(self, as_scene, as_main_assembly, as_project):
+        logger.debug("Flushing camera entity, num xform keys = %s", self.__as_camera.transform_sequence().size())
```
Comment (dictoon) [inline translators/cameras/final.py:65]: Another message that probably should be prefixed with `appleseed: `.

File: translators/cameras/final.py [hunk 6]
```diff
@@ -36,52 +37,63 @@
 
 
 class RenderCameraTranslator(Translator):
-    """
```
Comment (dictoon) [inline translators/cameras/final.py:39]: Isn't this comment relevant anymore?

File: translators/cameras/final.py [hunk 7]
```diff
@@ -197,24 +210,20 @@ def __thin_lens_camera_params(self, bl_scene, aspect_ratio, film_width, film_hei
...
-                                               aspect_ratio,
-                                               film_width,
-                                               film_height)
+        rotation_modify = asr.Matrix4d.make_rotation(asr.Vector3d(1.0, 0.0, 0.0), math.radians(-90.0))
 
-        cam_params.update({'projection_type': camera.data.appleseed.fisheye_projection_type})
+        matrix = rotation_modify * matrix
 
-        return cam_params
+        return asr.Transformd(matrix)
```
Comment (dictoon) [inline translators/cameras/final.py:229]: Add back end of line character.

File: translators/cameras/final.py [hunk 8]
```diff
@@ -197,24 +210,20 @@ def __thin_lens_camera_params(self, bl_scene, aspect_ratio, film_width, film_hei
...
             tex_name = f"{camera.data.appleseed.diaphragm_map.name_full}_inst"
             cam_params['diaphragm_map'] = tex_name
             del cam_params['diaphragm_blades']
 
         return cam_params
 
-    def __fisheye_camera_params(self, scene, aspect_ratio, film_width, film_height):
-        camera = self.bl_camera
+    def _convert_matrix(self, m):
+        matrix = asr.Matrix4d([m[0][0], m[0][1], m[0][2], m[0][3],
```
Comment (dictoon) [inline translators/cameras/final.py:220]: A good place to resurrect the explanation about Blender vs appleseed conventions?

**Comment by dictoon** [comment]
I've only had a cursory look so far, there are far too many changes to be able to do a real review. I suggest we discuss on Discord what the gist of this PR is, and maybe specific areas that I should focus on?

File: ui/objects.py (hunk 1 above)
Comment (jdent02) [reply ui/objects.py:165]: Limiting the assembly archive option to Empties.  Doesn't really make sense for the option to appear for meshes, curves or surfaces.

File: render/__init__.py (hunk 3 above)
Comment (jdent02) [reply render/__init__.py:77]: Yes.

File: render/__init__.py (hunk 4 above)
Comment (jdent02) [reply render/__init__.py:102]: I did do that but it looks like I missed this one.

File: translators/cameras/final.py (hunk 8 above)
Comment (jdent02) [reply translators/cameras/final.py:220]: I believe it is covered in the base Translator class.  I'll double check.

Comment (jdent02) [reply translators/cameras/final.py:220]: Nope it wasn't.  I added it back in
